iroha/src/sumeragi.rs

```diff
@@ -118,33 +116,36 @@ impl Sumeragi {
         }
     }
 
-    fn sort_peers(peers: &mut Vec<PeerId>, block_hash: Option<Hash>) {
-        peers.sort_by(|p1, p2| {
-            if let Some(block_hash) = block_hash {
-                let mut bytes_p1 = Vec::<u8>::from(p1);
-                bytes_p1.extend_from_slice(&block_hash);
-                let hash_p1 = crypto::hash(bytes_p1);
-                let mut bytes_p2 = Vec::<u8>::from(p2);
-                bytes_p2.extend_from_slice(&block_hash);
-                let hash_p2 = crypto::hash(bytes_p2);
-                let order = hash_p1.cmp(&hash_p2);
-                if order == Ordering::Equal {
-                    p1.address.cmp(&p2.address)
-                } else {
-                    order
-                }
-            } else {
-                p1.address.cmp(&p2.address)
-            }
-        });
+    pub fn sort_peers(peers: &mut Vec<PeerId>, block_hash: Option<Hash>) {
+        peers.sort_by(|p1, p2| p1.address.cmp(&p2.address));
+        if let Some(block_hash) = block_hash {
+            let mut rng = StdRng::from_seed(block_hash);
+            peers.shuffle(&mut rng);
+        }
     }
 
     pub fn sign_block(&self, block: Block) -> Result<Block, String> {
         self.validate_access(&[Role::Leader, Role::ProxyTail, Role::ValidatingPeer])?;
         Ok(block.sign(&self.public_key, &self.private_key)?)
     }
 
-    pub async fn on_block_created(&self, block: Block) -> Result<Block, String> {
+    pub async fn validate_and_store(
+        &self,
+        transactions: Vec<Transaction>,
+        wsv: Arc<Mutex<WorldStateView>>,
+    ) -> Result<(), String> {
+        let transactions = self.sign_transactions(transactions).await?;
+        let block = Kura::create_block(transactions, &*wsv.lock().await);
+        if self.sorted_peers.len() < 2 {
```

Comment:
maybe it can be a good practice to extract int into some named constant?

Reply:
It is a special case when we only have one peer, so to my mind using constant here will only make it harder to understand. Maybe I should just replace it with `.len()==1`. As right now it's impossible to create Sumeragi with 0 peers, it has to have itself at least.